lib/client.js

```diff
@@ -86,7 +86,8 @@ api.sendEvent = (event, peer, callback) => {
         'NetworkError', {
           httpStatusCode: res.statusCode,
           event,
-          peer
+          peer,
+          error: res.body
         }, typeof res.body === 'object' || new Error(res.body)));
```

Comment:
This `typeof` line is a bit strange, won't this set either `true` or an `Error`? Is that the intent? It seems like it could be fixed up and then we wouldn't need the additional `error` property that's being passed in the details.

Reply:
@dlongley in addition to the defect your pointed out, this pattern does not work well for two other reasons:
1. It actually does not work at all when doing `new Error`, but does work (as intended) with `new BedrockError`, but of course we can't make the assumption that the object returned adheres to the `BedrockError` API.
2. Dealing with the error like this puts the error details in `error.cause` apart from the other error details in `error.details` which is inconvenient and breaks with our current convention.

I fixed this up to *ONLY* include the error details (res.body) in `error.details` as per our convention.  If we are agreed that this pattern should be abolished, then other instances of its use will need to be removed as well.

Reply:
What you did looks good.